src/optick_impl.rs

```diff
@@ -1,13 +1,13 @@
 #[macro_export]
 macro_rules! scope {
-    ($name:expr) => {
+    ($name:expr) => {{
         $crate::optick::event!($name);
-    };
+    }};
     // NOTE: I've not been able to get attached data to work with optick
-    ($name:expr, $data:expr) => {
+    ($name:expr, $data:expr) => {{
```

Comment:
I'm not sure if we want the double {{ here, I think it will create a scope that immediately closes and it won't measure the intended code

Reply:
Yeah, I ran into this exact issue:
> Does it have the same problem you fixed in the tracy backend of not allowing multiple profiling!() macros in the same scope?

It fails on this snippet I added:
```rs
    // Test that using this macro multiple times in the same scope level will compile.
    {
        profiling::scope!("Outer scope");
        burn_time(5);
        profiling::scope!("Inner scope");
        burn_time(5);
    }
```

But I completely missed this issue with the scope ending early. :sweat_smile: 

For now, I can cfg out this snippet when optick is being used. I think in the future, we might need to mimic what the macro in `optick` is doing (but in a safer way) or try to upstream some changes there.

Reply:
Yeah, that sounds like a good way to go!

Reply:
I pushed the change